sunpy/util/xml.py

```diff
@@ -124,3 +123,66 @@ def get_node_text(node):
         else:
             raise NotTextNodeError
     return t
+
+def xml_comments_to_dict(xmlstring):
+    """
+    Parses all the XML comments provided in an XML string into a dictionary
+
+    .. warning::
+        This method does not support multiple values of comments for elements
+        of the same name but with different values. It always takes the last value.
+
+    Parameters
+    ----------
+    xmlstring : `str`
+        A `str` of xml content.
+
+    Returns
+    -------
+    `dict`:
+        The comments present in the string xml input as a dictionary.
+    Examples
+    --------
+    ::
+
+        <outer>
+            <inner comment = "One">one</inner>
+            <inner comment = "Two">two</inner>
+        </outer>
+
+    gives you the dict::
+
+        {u'inner': 'Two'}
```

Comment:
Can this not be a full runnable code example?

Reply:
I tried to create a runnable code example, seems like the `parseString()` function used to parse an XML string to an XML tree makes some kind of validation that prevents an XML tree like the one given below from being valid and is throwing a few errors. I guess the XML string would then become too long to accommodate all the validations.

```xml
        <outer>
            <inner comment = "One">one</inner>
            <inner comment = "Two">two</inner>
        </outer>
```